staging/src/k8s.io/kubectl/pkg/cmd/rollout/rollout_status_test.go

```diff
@@ -90,6 +91,35 @@ func TestRolloutStatus(t *testing.T) {
 	}
 }
 
+func TestRolloutStatusNoResources(t *testing.T) {
+	deploymentName := "deployment/nginx-deployment"
+	ns := scheme.Codecs.WithoutConversion()
+	tf := cmdtesting.NewTestFactory().WithNamespace("test")
+	tf.ClientConfigVal = cmdtesting.DefaultClientConfig()
+
+	info, _ := runtime.SerializerInfoForMediaType(ns.SupportedMediaTypes(), runtime.ContentTypeJSON)
+	encoder := ns.EncoderForVersion(info.Serializer, rolloutStatusGroupVersionEncoder)
+	tf.Client = &fake.RESTClient{
+		GroupVersion:         rolloutStatusGroupVersionEncoder,
+		NegotiatedSerializer: ns,
+		Client: fake.CreateHTTPClient(func(req *http.Request) (*http.Response, error) {
+			dep := &appsv1.ReplicaSet{}
+			dep.Name = deploymentName
+			body := io.NopCloser(bytes.NewReader([]byte(runtime.EncodeOrDie(encoder, dep))))
+			return &http.Response{StatusCode: http.StatusNotFound, Header: cmdtesting.DefaultHeader(), Body: body}, nil
```

Comment:
As far as I can see, unit test failure is due to this `http.StatusNotFound`

Reply:
@ardaguclu i missed removing a return statement in the command. Removed return r.Visit and am returning error after that function if it doesnt not run. otherwise the return will be in the Visit function. These tests are passing locally now. Thanks